### DISCUSSION TITLE
why `useRequest` in `onMounted` hook works well in 2.17.0, while not in 2.20.3

### DISCUSSION THREAD

**Discussion by phymo** [alovajs/alova#356] [Q&A]
I know that I should not write `useRequest` inside `onMounted`.
But I want to know which modification or commit make it not working in 2.20.3?

``` javascript
onMounted(() => {
  useRequest(getList());
});
```

**Comment by JOU-amjs** [comment]
Due to some reasons. it's will send request with async way in 2.20.x.
And the usage of the `useRequest` of yours is not the right way which not recommend extremely. It just use to send request and cannot use the states that returns in use hooks. [refer to the Rules of Hooks of react](https://legacy.reactjs.org/docs/hooks-rules.html).

so if you use `useRequest`. we recommend to use like below.
```html
<script setup>
const { loading, data, error } = useRequest(getList());
</script>
```

or you just wanna send request, you can do with a single method instance, like this.
```html
<script setup>
onMounted(async () => {
  await getList();
});
</script>
```

if there were many such this code, I recommend to downgrade to `2.19.x` or lower.
